api/lib/domain/usecases/link-user-to-organization-student-data.js

```diff
@@ -1,31 +1,38 @@
-const { NotFoundError, UserNotAuthorizedToAccessEntity } = require('../../domain/errors');
+const { NotFoundError, UserNotAuthorizedToAccessEntity, UserAlreadyLinkedToStudentError } = require('../../domain/errors');
 
 module.exports = async function linkUserToOrganizationStudentData({
   campaignCode,
   user,
   campaignRepository,
   studentRepository,
 }) {
-  let student;
-  const campaign = await campaignRepository.getByCode(campaignCode);
-  const organizationId = campaign.organizationId;
+  const userId = user.id;
 
-  if (user.id === null) {
+  if (userId === null) {
     throw new UserNotAuthorizedToAccessEntity('User is not part of the organization student list');
   }
 
+  const alreadyAssociatedStudent = await studentRepository.findOneByUserId({ userId });
+
+  if (alreadyAssociatedStudent) {
+    throw new UserAlreadyLinkedToStudentError('User is already student of another organization');
```

Comment:
`User is already student of another organization`, ce n'est pas ça qu'il se passe avec `UserAlreadyLinkedToStudentError ` mais plutôt `User is already linked to a student`, on n'a pas du tout mention de l'organization ici.

Reply:
Done.